eox_tenant/tenant_wise/proxies.py

```diff
@@ -126,43 +158,98 @@ def get_value_for_org(cls, org, val_name, default=None):
+        val_name and all organizations and returns the value for the given org.
+
+        DEPRECATED: use __get_value_for_org method instead.
+        """
+        warnings.warn(
+            "__deprecated_get_value_for_org, MySQL 5.7+ is required to use __get_value_for_org instead.",
+            DeprecationWarning
+        )
+
+        if not cls.has_configuration_set():
+            return default
+
+        cache_key = "org-value-{}-{}".format(org, val_name)
+        cached_value = cache.get(cache_key)
+
+        if cached_value:
+            return cached_value
+
+        result = None
+        tenant_config = TenantConfig.objects.values_list("lms_configs", flat=True)
+        microsite_config = Microsite.objects.values_list("values", flat=True)  # pylint: disable=no-member
 
         for config in chain(tenant_config, microsite_config):
             try:
-                current = json.loads(config[0])
-                org_filter = current.get("course_org_filter", {})
-            except IndexError:
+                if isinstance(config, six.string_types):
+                    config = json.loads(config)
+
+                org_filter = config.get("course_org_filter", {})
+            except AttributeError:
                 continue
 
             if org_filter:
                 if isinstance(org_filter, six.string_types):
                     org_filter = set([org_filter])
-                if org in org_filter:
-                    result = current.get(val_name, default)
-                    cls.set_key_to_cache(cache_key, result)
-                    return result
 
-        return default
+                for organization in org_filter:
+                    if org == organization and not result:
+                        result = config.get(val_name, default)
 
-    @classmethod
-    def has_configuration_set(cls):
-        """
-        We always require a configuration to function, so we can skip the query
-        """
-        return getattr(settings, "USE_EOX_TENANT", False)
+                    key = "org-value-{}-{}".format(organization, val_name)
+                    cls.set_key_to_cache(key, config.get(val_name, default))
+
+        if not result:
+            logger.warning("Tha value for %s and org %s has not been found", val_name, org)
+            cls.set_key_to_cache(cache_key, default)
+            result = default
+
+        return result
 
     @classmethod
-    def set_key_to_cache(cls, key, value):
+    def pre_load_values_by_org(cls, val_name):
         """
-        Stores a key value pair in a cache scoped to the thread
+        Save in cache the values for all the organizations in TenantConfig and Microsite models.
         """
-        cache.set(
-            key,
-            value,
-            EOX_TENANT_CACHE_KEY_TIMEOUT
-        )
+        tenant_config = TenantConfig.objects.filter(
+            lms_configs__has_key=u"course_org_filter",
```

Comment:
We are using the filter that requires mysql 5.7+, but we have no deprecated method for this. I would suggest that we try catch this queries If there is no mysql, then the preloading will not work, but at least is not going to break the execution of the request.